src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs

```diff
@@ -332,13 +335,110 @@ public override IntermediateNode Visit(TexlCallNode node, IRTranslatorContext co
+                            convertedNode = ReplaceBlankWithEmptyString(args[i]);
+                            break;
+                        default:
+                            convertedNode = args[i];
+                            break;
+                    }
+
+                    convertedArgs.Add(convertedNode);
+                }
+
+                return convertedArgs;
+            }
+
+            private static IntermediateNode ReplaceBlankWithZero(IntermediateNode arg)
+            {
+                var zeroNumLitNode = new NumberLiteralNode(IRContext.NotInSource(FormulaType.Number), 0);
+                return ReplaceBlankWithHelper(arg, zeroNumLitNode, IRContext.NotInSource(FormulaType.Number));
+            }
+
+            private static IntermediateNode ReplaceBlankWithEmptyString(IntermediateNode arg)
+            {
+                var emptyStringNode = new TextLiteralNode(IRContext.NotInSource(FormulaType.String), string.Empty);
+                return ReplaceBlankWithHelper(arg, emptyStringNode, IRContext.NotInSource(FormulaType.String));
+            }
+
+            private static IntermediateNode ReplaceBlankWithHelper(IntermediateNode arg, IntermediateNode defaultNode, IRContext irContext)
+            {
+                var isBlankNode = new CallNode(IRContext.NotInSource(FormulaType.Boolean), BuiltinFunctionsCore.IsBlank, arg);
+               
+                if (IsNodeBlankToDefault(arg, defaultNode))
+                {
+                    return arg;
+                }
+
+                var convertedNode = new CallNode(irContext, BuiltinFunctionsCore.If, isBlankNode, defaultNode, arg);
+                return convertedNode;
+            }
+
+            private static IntermediateNode ConcatenateArgs(IntermediateNode arg1, IntermediateNode arg2)
+            {
+                var concatenateArgs = new List<IntermediateNode>();
+                foreach (var arg in new[] { arg1, arg2 })
+                {
+                    // if arg is call node to Concatenate unpack it, and pass it as arg to outer Concatenate
+                    if (arg is CallNode maybeConcatenate && maybeConcatenate.Function is ConcatenateFunction concatenateFunction)
+                    {
+                        foreach (var argC in maybeConcatenate.Args)
+                        {
+                            concatenateArgs.Add(ReplaceBlankWithEmptyString(argC));
+                        }
+                    }
+                    else
+                    {
+                        concatenateArgs.Add(ReplaceBlankWithEmptyString(arg));
+                    }
+                }
+
+                var concatenatedNode = new CallNode(IRContext.NotInSource(FormulaType.String), BuiltinFunctionsCore.Concatenate, concatenateArgs);
+                return concatenatedNode;
+            }
+
+            private static bool IsNodeBlankToDefault(IntermediateNode arg, IntermediateNode defaultNode)
```

Comment:
what is this doing?

Reply:
This Checks if Node is already wrapped in BlankToDefault.